manifests/charts/istio-cni/templates/daemonset.yaml

```diff
@@ -89,18 +89,14 @@ spec:
               path: /readyz
               port: 8000
           securityContext:
-            privileged: true # always requires privilege to be useful (install node plugin, etc)
+            privileged: false
             runAsGroup: 0
             runAsUser: 0
             runAsNonRoot: false
             # Both ambient and sidecar repair mode require elevated node privileges to function.
-            # But we don't need _everything_ in `privileged`, so drop+readd capabilities based on feature.
-            # privileged is redundant with CAP_SYS_ADMIN
-            # since it's redundant, hardcode it to `true`, then manually drop ALL + readd granular
-            # capabilities we actually require
+            # But we don't need _everything_ in `privileged`, so explicitly set it to false and
```

Comment:
This is a regrettable feature of the `securityContext.privileged` API in K8S. It's a bad API. You can set it to `false` but still have root on the node - which is what we had before. Which was massively confusing to end-users.

We are self-granting `CAP_SYS_ADMIN` and running as root, which means this pod _is_ completely privileged as root on the node in actual fact, regardless of the value of `securityContext.privileged`.

This is why, for instance, [in the k8s Pod Security Standards, `securityContext.privileged` and `securityContext.capabilities` are always lockstepped](https://kubernetes.io/docs/concepts/security/pod-security-standards/#baseline)

As long as we have CAP_SYS_ADMIN `istio-cni` is absolutely a privileged container, any way you measure it, regardless of the value of `securityContext.privileged`.

And setting this to `false` while keeping the `capabilities.add.SYS_ADMIN` still doesn't allow us to run under a different PSS profile anyway - [we are still stuck with `privileged`](https://kubernetes.io/docs/concepts/security/pod-security-standards/#privileged)

So I don't think setting this to `false` actually does anything but mislead people and get us past not-very-smart manifest-based pod admission gates.

Reply:
I totally agree setting privileged to false is confusing and there is little difference between the options when setting CAP_SYS_ADMIN.

My main motivation is allowing user to explicitly see the permissions being added and be able to evaluate the risk of each. Setting privileged to true and then dropping all capabilities still leaves elevated privileges (in addition to the capabilities we currently add back). I think this scenario is also confusing to the user.

Reply:
> Setting privileged to true and then dropping all capabilities still leaves elevated privileges (in addition to the capabilities we currently add back). I think this scenario is also confusing to the user.

+1 on this, it's a good point and I don't object to hardcoding it to `false`. 

In general, I want to hew to the published K8S Pod Security Standards  - as per those, this pod is _always_ in the Privileged profile, and _never_ in the Baseline or Restricted profile.

I only object to making it a toggle to fit an unspecified "compliance" profile that isn't the K8S Pod Security Standard - because any compliance policy that would reject `securityContext.privileged: true` but not reject `securityContext.capabilities.CAP_SYS_ADMIN` is explicitly broken from the perspective of that standard.

Reply:
> I only object to making it an exposed toggle to fit an unspecified "compliance" profile that isn't the K8S Pod Security Standard - because any compliance policy that would reject securityContext.privileged: true but not reject securityContext.capabilities.CAP_SYS_ADMIN is explicitly broken from the perspective of that standard.

+1